All executor feature.

jrpalmer opened the pull request:
Proposing a small feature that will dynamically set weight to the number of executors. Example: I wanted to be able to use a templated build job that can run on any node, with any number of executors. When this particular job builds it can be on the only thing building as it does cleanup that would cause any other build to fail. This plug in worked great, until I had nodes with different number of executors. Take a look, let me know if anything needs correcting.

Commit:
Added functionality for the ability to use all executors. This is to handle the case when I only want one build occuring at a given time.

Commit:
Minor update to Help copy.

buildhive commented:
[Jenkins » heavy-job-plugin #4](https://buildhive.cloudbees.com/job/jenkinsci/job/heavy-job-plugin/4/) FAILURE
Looks like there's a problem with this pull request
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

jrpalmer commented:
The build failed due to cause : Execution default-apt-compile of goal org.jvnet.hudson.tools:maven-hpi-plugin:1.54:apt-compile failed: A required class was missing while executing org.jvnet.hudson.tools:maven-hpi-plugin:1.54:apt-compile: com/sun/mirror/apt/AnnotationProcessorFactory 

The plugin builds locally for me, not sure why this fails on build hive.

Commit:
Updated jenkins dependency version and implemented required methods for Executable.

buildhive commented:
[Jenkins » heavy-job-plugin #5](https://buildhive.cloudbees.com/job/jenkinsci/job/heavy-job-plugin/5/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

Commit:
Added radio blocks so you can only set the weight in one place.

buildhive commented:
[Jenkins » heavy-job-plugin #6](https://buildhive.cloudbees.com/job/jenkinsci/job/heavy-job-plugin/6/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

Commit:
Fixed an issue in the jelly config.

buildhive commented:
[Jenkins » heavy-job-plugin #7](https://buildhive.cloudbees.com/job/jenkinsci/job/heavy-job-plugin/7/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

Commit:
Minor refactor. Realized I had getNumExecutors at Computer level.

buildhive commented:
[Jenkins » heavy-job-plugin #8](https://buildhive.cloudbees.com/job/jenkinsci/job/heavy-job-plugin/8/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

Commit:
Fixed an NPE error when running the same job multiple times at the same
time. For now, I just return the estimated duration as 0. Will
investigate what to pass later on.

buildhive commented:
[Jenkins » heavy-job-plugin #9](https://buildhive.cloudbees.com/job/jenkinsci/job/heavy-job-plugin/9/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

jrpalmer commented:
This isn't working properly with multiple nodes. Going to investigate more and resubmit.
